### PR TITLE
RM - Message Structure Design

### PR DESCRIPTION
**Pull request by Diama1**
#604

### CODE REVIEW

File: client/src/elm/Pages/MessagingCenter/View.elm [hunk 1]
```diff
@@ -512,17 +512,25 @@ viewResilienceMessage language nurseId nurse model ( messageId, message ) =
 
 viewIntroductionMessage : Language -> Nurse -> ResilienceMessageOrder -> ( List (Html Msg), List (Html Msg) )
 viewIntroductionMessage language nurse order =
+    let
+        introductionCategoryHeader =
```
Comment (anvmn) [inline client/src/elm/Pages/MessagingCenter/View.elm]: Category header should be added at `viewResilienceMessage()`

**Review by anvmn — CHANGES_REQUESTED** [review]

File: client/src/elm/Translate.elm [hunk 2]
```diff
@@ -1289,6 +1289,13 @@ type TranslationId
     | Reports
     | RecentAndUpcomingGroupEncounters
     | ReportCompleted { pending : Int, completed : Int }
+    | ResilienceCategoryIntroductionHeader
```
Comment (anvmn) [inline client/src/elm/Translate.elm]: Instead all these, translate `ResilienceCategory`.

**Review by anvmn — CHANGES_REQUESTED** [review]

**Comment by Diama1** [comment]
RM

File: .ddev/commands/web/client-install [hunk 3]
```diff
@@ -3,6 +3,7 @@
 ## Description: Run robo commands inside the web container for the client app.
 ## Usage: client-install
 ## Example: "ddev client-install".
+export PATH=~/.nvm/versions/node/v9.0.0/bin:$PATH;
```
Comment (anvmn) [inline .ddev/commands/web/client-install]: Should not be part of this PR. Remove.

File: .ddev/commands/web/gulp [hunk 4]
```diff
@@ -28,5 +28,5 @@ then
   cp ../../eheza-infra/elm/Config.Deploy.elm ${CLIENT_PATH}/src/elm/
 
 fi
-
+export PATH=~/.nvm/versions/node/v9.0.0/bin:$PATH;
```
Comment (anvmn) [inline .ddev/commands/web/gulp]: Should not be part of this PR. Remove.

**Review by anvmn — CHANGES_REQUESTED** [review]

**Comment by anvmn** [comment]
@Diama1 

What's the status here?

**Comment by Diama1** [comment]
> @Diama1
> 
> What's the status here?

Working on the giving colors to different tabs, favorite and the read messages

**Comment by Diama1** [comment]
rmFin

**Comment by anvmn** [comment]
@Diama1 

From your demo, I see 2 problems:

1. Category icons should be of common height, and aligned with category title.
2. Format of date is not as requested.

**Comment by Diama1** [comment]
Screenshot from 2023-02-15 16-09-28

**Comment by Diama1** [comment]
Screenshot from 2023-02-15 16-09-28

**Comment by anvmn** [comment]
@Diama1 is this ready for review?

File: .ddev/commands/web/gulp (hunk 4 above)
Comment (Diama1) [reply .ddev/commands/web/gulp]: > @Diama1 is this ready for review?

Yeah, it took long for the test to pass but now it ready for review

**Comment by Diama1** [comment]
> @Diama1 is this ready for review?

Yeah, it took long for the test to pass but now it ready for review

File: client/src/assets/scss/_new.scss [hunk 5]
```diff
@@ -4631,27 +4631,115 @@ div.page-activity.messaging-center {
...
 
-    .title-wrapper .title {
+    .icon-options {
+      margin-top: 20px;
+      padding: 20px;
+      background: url(#{$img-path}icon-options.svg) center center no-repeat;
+      background-size: 32px 30px;
+      height: 32px;
+      width: 30px;
+      }
```
Comment (anvmn) [inline client/src/assets/scss/_new.scss]: format

File: client/src/assets/scss/_new.scss [hunk 6]
```diff
@@ -4631,27 +4631,115 @@ div.page-activity.messaging-center {
...
     margin-bottom: 30px;
 
-    .title,
-    .title-wrapper {
+    .title-wrapper, .header {
       background-color: $color-velvet;
       color: $color-white;
+
+    }
+    .header {
```
Comment (anvmn) [inline client/src/assets/scss/_new.scss:4639]: add empty line above

File: client/src/assets/scss/_new.scss [hunk 7]
```diff
@@ -4631,27 +4631,115 @@ div.page-activity.messaging-center {
...
+
+    .icon-connecting {
+      background: url(#{$img-path}icon-connecting-red.svg) center center no-repeat;
+      background-size: 30px 30px;
+      height: 30px;
+      margin-right: 5px;
+      margin-left: -1px;
+      margin-top: -1px;
+      width: 30px;
+      }
```
Comment (anvmn) [inline client/src/assets/scss/_new.scss]: format

File: client/src/assets/scss/_new.scss [hunk 8]
```diff
@@ -4631,27 +4631,115 @@ div.page-activity.messaging-center {
...
+
+    .icon-connecting.blue {
+      background: url(#{$img-path}icon-connecting-blue.svg) center center no-repeat;
+      }
```
Comment (anvmn) [inline client/src/assets/scss/_new.scss]: format

File: client/src/assets/scss/_new.scss [hunk 9]
```diff
@@ -4631,27 +4631,115 @@ div.page-activity.messaging-center {
...
+
+    .icon-connecting.purple {
+        background: url(#{$img-path}icon-connecting-purple.svg) center center no-repeat;
+      }
```
Comment (anvmn) [inline client/src/assets/scss/_new.scss]: format

File: client/src/elm/Pages/MessagingCenter/View.elm [hunk 10]
```diff
@@ -427,40 +428,50 @@ viewTabs language model =
 viewResilienceMessage : Language -> NurseId -> Nurse -> Model -> ( ResilienceMessageId, ResilienceMessage ) -> Html Msg
 viewResilienceMessage language nurseId nurse model ( messageId, message ) =
     let
-        ( extraClass, ( head, body ) ) =
+        messageCategory =
+            span [ class "category-header" ] [ text <| translate language <| Translate.ResilienceCategory message.category ]
+
+        ( extraClass, header, ( head, body ) ) =
```
Comment (anvmn) [inline client/src/elm/Pages/MessagingCenter/View.elm]: If `header` is `messageCategory` for all, then we don't need to resolve it according to `message.category`.
Make it `( extraClass, ( head, body ) ) = ` ...

File: client/src/elm/Pages/MessagingCenter/View.elm [hunk 11]
```diff
@@ -476,23 +487,49 @@ viewResilienceMessage language nurseId nurse model ( messageId, message ) =
         messageClickedAction =
             ResilienceMessageClicked nurseId messageId message updateTimeRead
 
+        sentDate =
+            case nurse.resilienceProgramStartDate of
+                Just date ->
```
Comment (anvmn) [inline client/src/elm/Pages/MessagingCenter/View.elm]: use Maybe.map

File: client/src/elm/Pages/MessagingCenter/View.elm [hunk 12]
```diff
@@ -427,40 +428,50 @@ viewTabs language model =
...
             case message.category of
                 ResilienceCategoryIntroduction ->
                     ( "introduction"
+                    , messageCategory
                     , viewIntroductionMessage language nurse message.order
                     )
 
                 ResilienceCategoryGrowth ->
-                    ( "growth"
+                    ( "growth "
```
Comment (anvmn) [inline client/src/elm/Pages/MessagingCenter/View.elm]: Remove space at the end.
Same for all options bellow.

File: client/src/elm/Pages/MessagingCenter/View.elm [hunk 13]
```diff
@@ -476,23 +487,49 @@ viewResilienceMessage language nurseId nurse model ( messageId, message ) =
...
         title =
             let
                 plainTitle =
-                    div
-                        [ class "title"
-                        , onClick messageClickedAction
+                    div [ class <| "header " ++ titleWrapperClass, onClick messageClickedAction ]
+                        [ i [ class <| "icon-" ++ extraClass ++ titleWrapperClass ] []
+                        , header
+                        , span [ class "date-sent" ]
```
Comment (anvmn) [inline client/src/elm/Pages/MessagingCenter/View.elm:503]: If `sentDate` is `Nothing` we should not be showing this whole `span`

File: client/src/assets/scss/_new.scss [hunk 14]
```diff
@@ -4631,27 +4631,115 @@ div.page-activity.messaging-center {
...
+    .title-wrapper, .header {
       background-color: $color-velvet;
       color: $color-white;
+
+    }
+    .header {
+      display: flex;
+      flex-flow: row wrap;
       padding: 20px;
+      &.blue {
```
Comment (anvmn) [inline client/src/assets/scss/_new.scss:4644]: add empty line above

File: client/src/assets/scss/_new.scss [hunk 15]
```diff
@@ -4631,27 +4631,115 @@ div.page-activity.messaging-center {
...
+        background-color: $color-primary;
+      }
+      &.purple {
```
Comment (anvmn) [inline client/src/assets/scss/_new.scss:4648]: add empty line above

File: client/src/assets/scss/_new.scss [hunk 16]
```diff
@@ -4631,27 +4631,115 @@ div.page-activity.messaging-center {
...
+      flex: 1;
+    }
+
+    .header .title {
       padding: 0;
     }
 
+    .date-sent {
+      flex-basis: 100%;
+      width: 0;
```
Comment (anvmn) [inline client/src/assets/scss/_new.scss]: Why is this needed?

**Review by anvmn — CHANGES_REQUESTED** [review]
Selection_1586
Selection_1585

File: client/src/assets/scss/_new.scss (hunk 16 above)
Comment (Diama1) [reply client/src/assets/scss/_new.scss]: > Why is this needed?

I used flex-basis for the date to come on the new line line

**Comment by Diama1** [comment]
rm1

File: client/src/elm/Pages/MessagingCenter/View.elm [hunk 17]
```diff
@@ -476,23 +480,44 @@ viewResilienceMessage language nurseId nurse model ( messageId, message ) =
...
+        titleWrapperClass =
+            case model.activeTab of
+                TabUnread ->
+                    ""
+
+                TabFavorites ->
+                    " purple"
+
+                _ ->
+                    " blue"
```
Comment (anvmn) [inline client/src/elm/Pages/MessagingCenter/View.elm]: Remove space

File: client/src/elm/Pages/MessagingCenter/View.elm [hunk 18]
```diff
@@ -476,23 +480,44 @@ viewResilienceMessage language nurseId nurse model ( messageId, message ) =
...
+        sentDate =
+            Maybe.map (Date.add Days message.displayDay) nurse.resilienceProgramStartDate
+
+        titleWrapperClass =
+            case model.activeTab of
+                TabUnread ->
+                    ""
+
+                TabFavorites ->
+                    " purple"
```
Comment (anvmn) [inline client/src/elm/Pages/MessagingCenter/View.elm]: Remove space

File: client/src/elm/Pages/MessagingCenter/View.elm (hunk 13 above)
Comment (anvmn) [reply client/src/elm/Pages/MessagingCenter/View.elm:503]: ??

File: client/src/assets/scss/_new.scss (hunk 5 above)
Comment (anvmn) [reply client/src/assets/scss/_new.scss]: ??

File: client/src/assets/scss/_new.scss [hunk 19]
```diff
@@ -4631,27 +4631,116 @@ div.page-activity.messaging-center {
...
+      font-size: 24px;
+      flex: 1;
+    }
+
+    .header .title {
       padding: 0;
     }
 
+    .date-sent {
+      flex-basis: 100%;
```
Comment (anvmn) [inline client/src/assets/scss/_new.scss]: Is this needed at all?

**Review by anvmn — CHANGES_REQUESTED** [review]

**Review by anvmn — APPROVED** [review]
